### PR TITLE
network: add nlmon support

### PR DESCRIPTION
**Pull request by yuwata**
~~This does not work...~~

~~@ssahani Could you comment on this? I have no idea why this does not work.~~

### CODE REVIEW

**Comment by ssahani** [comment]
Please do not generate the MAC
```
 src/network/netdev/netdev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/netdev/netdev.c b/src/network/netdev/netdev.c
index bac2813366..5d6fbcd708 100644
--- a/src/network/netdev/netdev.c
+++ b/src/network/netdev/netdev.c
@@ -737,7 +737,7 @@ int netdev_load_one(Manager *manager, const char *filename) {
         if (!netdev->filename)
                 return log_oom();

-        if (!netdev->mac && !IN_SET(netdev->kind, NETDEV_KIND_VLAN, NETDEV_KIND_BRIDGE)) {
+        if (!netdev->mac && !IN_SET(netdev->kind, NETDEV_KIND_VLAN, NETDEV_KIND_BRIDGE, NETDEV_KIND_NLMON)) {
                 r = netdev_get_mac(netdev->ifname, &netdev->mac);
                 if (r < 0)
                         return log_netdev_error_errno(netdev, r,
-- 
2.21.0
```

**Comment by yuwata** [comment]
@ssahani Thanks a lot!

**Comment by yuwata** [comment]
Updated. PTAL.

File: src/network/netdev/netdev.h [hunk 1]
```diff
@@ -131,6 +131,9 @@ typedef struct NetDevVTable {
 
         /* verify that compulsory configuration options were specified */
         int (*config_verify)(NetDev *netdev, const char *filename);
+
+        /* Generate MAC address or not When MACAddress= is not specified. */
+        bool do_not_generate_mac;
```
Comment (ssahani) [inline src/network/netdev/netdev.h]: I think this should be ```generate_mac``` .

**Comment by ssahani** [comment]
Sorry for late reply. I did not have a laptop. Now I got it and got configured for devel environment .

File: src/udev/udev-rules.c.bak [hunk 2]
```diff
@@ -0,0 +1,2639 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
```
Comment (ssahani) [inline src/udev/udev-rules.c.bak]: What is this .bak file ?

File: src/network/netdev/tuntap.c [hunk 3]
```diff
@@ -161,4 +162,5 @@ const NetDevVTable tap_vtable = {
         .done = tuntap_done,
         .create = netdev_create_tuntap,
         .create_type = NETDEV_CREATE_INDEPENDENT,
+        .do_not_generate_mac = true,
```
Comment (ssahani) [inline src/network/netdev/tuntap.c]: Please drop this. we can set mac for tap device.

Comment (yuwata) [reply src/network/netdev/tuntap.c]: MACAddress= for tun or tap devices are set by .network file, not .netdev file. So, I think the generated mac address is simply ignored. No? Am I missing something?

File: src/udev/udev-rules.c.bak (hunk 2 above)
Comment (yuwata) [reply src/udev/udev-rules.c.bak]: haha. Will drop it.

File: src/network/netdev/tuntap.c (hunk 3 above)
Comment (ssahani) [reply src/network/netdev/tuntap.c]: Yes then it's not required at all. By default it's false no ? I mean ```generate_mac``` .

Comment (yuwata) [reply src/network/netdev/tuntap.c]: Ah, I see. Will update.

**Comment by yuwata** [comment]
@ssahani Thank you for your review. I've force-pushed a revised version. In the revised version, now the new flag is renamed to `generate_mac`. PTAL.

**Comment by ssahani** [comment]
LGTM.

**Pull request merged**